tv_apps/fling-player/test/unit/connector_test.js

```diff
@@ -0,0 +1,301 @@
+      connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus();
+    });
+
+    test('should report status of error', function () {
+      msg = castingMsgTemplate.get().statusError;
+      msg.seq = connector._msgSeq;
+      connector.reportStatus(msg.status, {
+        time : msg.time,
+        error : msg.error
+      });
+      assertReportStatus();
+    });
+
+    test('should not report status of unknown', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      msg.seq = connector._msgSeq;
+      msg.status = 'unknown';
+      expect(() => connector.reportStatus(msg.status, { time : msg.time }))
+            .to.throw(/Ilegal status/);
+    });
+
+    test('should not report status without time', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      msg.seq = connector._msgSeq;
+      expect(() => connector.reportStatus(msg.status, {}))
+            .to.throw(/Ilegal time/);
+    });
+  });
+
+  suite('handle remote request message', function () {
+
+    var spyOnLoadRequest, spyOnPlayRequest, spyOnSeekRequest, spyOnPauseRequest;
+
+    function assertHandled() {
+      assert.isTrue(
+        spy.withArgs(msg).calledOnce, 'Not hanlding remote message'
+      );
+      assert.equal(
+        connector._lastSeq, msg.seq, 'Last received msg seq not updated'
+      );
+    }
+
+    function assertNotHandled() {
+      assert.isFalse(
+        spy.withArgs(msg).calledOnce, 'Should not handle remote message'
+      );
+      assert.notEqual(
+        connector._lastSeq, msg.seq, 'Should not update last received msg seq'
+      );
+    }
+
+    function assertRequestCallCount(
+      loadRequestCount, playRequestCount, seekRequestCount, pauseRequestCount
+    ) {
+      assert.equal(
+        spyOnLoadRequest.callCount, loadRequestCount,
+        'Wrong load request count'
+      );
+      assert.equal(
+        spyOnPlayRequest.callCount, playRequestCount,
+        'Wrong play request count'
+      );
+      assert.equal(
+        spyOnSeekRequest.callCount, seekRequestCount,
+        'Wrong seek request count'
+      );
+      assert.equal(
+        spyOnPauseRequest.callCount, pauseRequestCount,
+        'Wrong pause request count'
+      );
+    }
+
+    setup(function () {
+      spy = sinon.spy(connector, 'handleRemoteMessage');
+      spyOnLoadRequest = sinon.spy();
+      spyOnPlayRequest = sinon.spy();
+      spyOnSeekRequest = sinon.spy();
+      spyOnPauseRequest = sinon.spy();
+      connector.on('loadRequest', spyOnLoadRequest);
+      connector.on('playRequest', spyOnPlayRequest);
+      connector.on('seekRequest', spyOnSeekRequest);
+      connector.on('pauseRequest', spyOnPauseRequest);
+    });
+
+    test('should handle load message', function () {
+      msg = castingMsgTemplate.get().load;
+      msg.seq = connector._lastSeq + 1;
+      presentation._controller.castMsg(msg);
+      assertHandled();
+      assertRequestCallCount(1, 0, 0, 0);
```

Comment:
It's just more difficult to understand than just write the `assert.equal` directly.